include/aws/io/private/pkcs11_private.h

```diff
@@ -15,5 +17,47 @@
 
 AWS_EXTERN_C_BEGIN
 
+/**
+ * Find the slot that meets all criteria:
+ * - has a token
+ * - if match_slot_id is non-null, then slot IDs must match
+ * - if match_token_label is non-null, then labels must match
+ * The function fails unless it finds exactly one slot meeting all criteria.
+ */
+AWS_IO_API
+int aws_pkcs11_lib_find_slot_with_token(
+    struct aws_pkcs11_lib *pkcs11_lib,
+    const unsigned long *match_slot_id,
+    const struct aws_string *match_token_label,
+    unsigned long *out_slot_id);
+
+AWS_IO_API
+int aws_pkcs11_lib_open_session(
+    struct aws_pkcs11_lib *pkcs11_lib,
+    unsigned long slot_id,
+    unsigned long *out_session_handle);
+
+AWS_IO_API
+void aws_pkcs11_lib_close_session(struct aws_pkcs11_lib *pkcs11_lib, unsigned long session_handle);
```

Comment:
because the calling code is shutting things down, and wouldn't be able to do anything differently if an error was reported.

I'm just not a fan of returning errors if the calling code can't reasonably do anything in response